tests/out_test.go

```diff
@@ -55,6 +59,31 @@ var _ = Describe("Out", func() {
 		Expect(session.Err).To(gbytes.Say(dockerd(`.*--data-root /scratch/docker.*`)))
 	})
 
+	Context("when build arguments are provided", func() {
+		It("passes the arguments correctly to the docker daemon", func() {
+			session := put(map[string]interface{}{
+				"source": map[string]interface{}{
+					"repository": "test",
+				},
+				"params": map[string]interface{}{
+					"build": "/docker-image-resource/tests/fixtures/build",
+					"build_args": map[string]string{
+						"arg1": "arg with space",
+						"arg2": "arg with\nnewline",
+						"arg3": "normal",
+					},
+				},
+			})
+
+			Expect(session.Err).To(gbytes.Say(dockerarg(`--build-arg`)))
+			Expect(session.Err).To(gbytes.Say(dockerarg(`arg1=arg with space`)))
+			Expect(session.Err).To(gbytes.Say(dockerarg(`--build-arg`)))
+			Expect(session.Err).To(gbytes.Say(dockerarg(`arg2=arg with\nnewline`)))
```

Comment:
is this correct? since it's a backtick-quoted string, it'll contain literally `"\\n"`, not a linebreak, so this is asserting that it sees `"DOCKER ARG: arg2=arg with\\nnewline"`. Shouldn't that have an actual linebreak, not `\n`?

or does the docker cli interpret that character as a linebreak when parsing the build args?

Reply:
Good question.

It turns out both:
```
        Expect(session.Err).To(gbytes.Say(dockerarg(`arg2=arg with\nnewline`)))
```
and
```
        Expect(session.Err).To(gbytes.Say(dockerarg(`arg2=arg with
  newline`)))
```
pass the test. From investigating the output, it definitely is receiving the argument with a newline. I wonder if `gbytes.Say` is causing it to behave unexpectedly.

Reply:
That appears to be the case. When I change the assertion to:
```
       Expect(string(session.Err.Contents())).To(ContainSubstring(dockerarg(`arg2=arg with\nnewline`)))

```
it fails as expected, while the following passes:
```
       Expect(string(session.Err.Contents())).To(ContainSubstring(dockerarg(`arg2=arg with
 newline`)))
```

`gbytes.Say` converts the string to a regex, and `\n` matches the newline character.

Reply:
That being said, I think the assertion in the test is still correct.